shap/explainers/_tree.py

```diff
@@ -53,6 +54,41 @@
 }
 
 
+def _safe_check_tree_instance_experimental(tree_instance: Any) -> None:
+    """
+    This function checks if a tree instance has an experimental integration with shap TreeExplainer class.
+
+    To add experimental message support for your library add package name and its versions
+    verified to be used with shap to the 'experimental' dictionary below.
+
+    Parameters
+    ----------
+    tree_instance: object, tree instance from an external library
+    """
+    experimental = {
+        "causalml": "0.15.3",
+    }
+
+    safe_instance = None
+    if hasattr(tree_instance, "__class__"):
+        if hasattr(tree_instance.__class__, "__module__"):
+            safe_instance = tree_instance
+
+    if safe_instance:
+        library = safe_instance.__class__.__module__.split(".")[0]
+        if experimental.get(library):
+            warnings.warn(
+                f"You are using experimental integration with {library}. "
+                f"The {library} support is verified for the following versions: {experimental.get(library)}.",
```

Comment:
Can we make clear that we do not commit ourselves to supporting this long term
```suggestion
                f"The {library} support is verified for the following versions: {experimental.get(library)}. "
                 f"As experimental functionality, this integration may be removed or significantly changed in future releases without following semantic versioning. Use in production systems at your own risk.",
```